yugabyte/yugabyte-db
# [#7706] [Platform] [Backport-2.4] Edit backup config credentials.

**Description:**
Backup config security credentials can change over time. The platform should support editing the security credentials even if there are existing backups associated with a Backup Config. When security credentials expire, the Platform will fail to take new backups and also fail to delete old backups.

This change should be implemented for all three storage configs:

AWS: Allow editing Access Key/Secret, also toggling between IAM and AccessKey
GCP: Allow editing GCS Credentials [Do not allow editing bucket name]
Azure: Allow editing Azure SAS token [Do not allow editing Container URL]

**Test Plan:**
1. Hit the config button from the SideNav.
2. Click on Backup Tab and then hit the amazon s3 tab.
3. Create a backup config for s3.
4. Once it got created then you'll see the Edit configuration button next to the Delete configuration button.
5. Click on the Edit Configuration button.
6. AWS access credentials should be editable.
7. Do the same for the rest of the configs.

sbt test:

```[info] Test com.yugabyte.yw.cloud.UniverseResourceDetailsTest.testAddPriceWithRemovingOneNode started
[info] Test run finished: 0 failed, 0 ignored, 7 total, 1.182s
[error] Failed: Total 1130, Failed 1, Errors 0, Passed 1125, Skipped 4
[error] Failed tests:
[error] 	com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 616 s, completed 21 May, 2021 11:43:48 AM
mahebhat@USBGOMAHEBHAT1 managed % sbt "test-only com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest"
[info] Loading project definition from /Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/project
Installing the s3:// URLStreamHandler via java.net.URL.setURLStreamHandlerFactory
Creating a new Ivy URLHandlerDispatcher to handle s3:// URLs
[info] Set current project to yugaware (in build file:/Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/)
[info] Instrumenting 1087 classes to /Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/target/scala-2.11/jacoco/instrumented-classes
[info] Test run started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgradeWithTServerGFlags started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testCertUpdateNonRolling started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsNonRollingTServerOnlyUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testSoftwareNonRollingUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsNonRollingUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgradeWithSameMasterFlags started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testSoftwareUpgradeWithoutVersion started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgradeWithMasterGFlags started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testRemoveFlags started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testSoftwareUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsNonRollingMasterOnlyUpgrade started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgradeWithEmptyFlags started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testRollingRestart started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testCertUpdateRolling started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testSoftwareUpgradeWithSameVersion started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testCertUpdateFailureDifferentCerts started
[info] Test com.yugabyte.yw.commissioner.tasks.UpgradeUniverseTest.testGFlagsUpgradeWithSameTserverFlags started
[info] Test run finished: 0 failed, 0 ignored, 18 total, 28.826s
[info] Passed: Total 18, Failed 0, Errors 0, Passed 18
[success] Total time: 32 s, completed 21 May, 2021 11:45:22 AM
mahebhat@USBGOMAHEBHAT1 managed % git branch
  6945-Limit-Task-Queries
  7594-Delete-Backup-Task-History
* 7706-Edit-Config-2.4
  7706-Edit-Config-2.6
  master
mahebhat@USBGOMAHEBHAT1 managed % ```
